python/ray/autoscaler/autoscaler.py

```diff
@@ -389,6 +390,39 @@ def recover_if_needed(self, node_id, now):
         updater.start()
         self.updaters[node_id] = updater
 
+    def _get_node_specific_key(self, node_id, key):
+        node_tags = self.provider.node_tags(node_id)
+        if TAG_RAY_USER_NODE_TYPE in node_tags:
+            node_type = node_tags[TAG_RAY_USER_NODE_TYPE]
+            if node_type not in self.available_node_types:
+                raise ValueError(
+                    "Unknown node type tag: {}.".format(node_type))
+            node_specific_config = self.available_node_types[node_type]
+            if key in node_specific_config:
+                return node_specific_config[key].copy()
+        return None
+
+    def _get_worker_initialization_commands(self, node_id):
+        from ray.autoscaler.docker import docker_pull_if_needed
+        default = self.config["initialization_commands"]
+        node_specific = self._get_node_specific_key(node_id,
+                                                    "initialization_commands")
+        if node_specific:
+            docker_pull_if_needed(self.config, node_specific)
```

Comment:
I thought we were going to do these rewrites at parse time? We shouldn't need this here in that case.